### PR TITLE
auth: add setting workaround-11804-max-chunk-records

### PR DESCRIPTION
**Pull request by zeha**
### Short description

Workaround for https://github.com/PowerDNS/pdns/issues/11804. Defaults to previously hard-coded value of 100.

I will add docs.

### Checklist

I have:
- [ ] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by mind04** [comment]
IMHO this option should be a switch. There is only one 99.9% safe value (1) and the default (100) is working well the majority of installations. Tuning this to other values make no sense and will most likely result in more problems.

Ps 99.9% as it is still possible to overfill a chunk when there is no space left for the added RRSIG.

**Comment by Habbie** [comment]
> Ps 99.9% as it is still possible to overfill a chunk when there is no space left for the added RRSIG(s).

In which case your data is un-serveable anyway :)

**Comment by klaus-nicat** [comment]
Interesting behavior for zones signed by PDNS (not presigned): with maxchunk(1) there is always 1 answer RR + TSIG per message. RRSIGs are a DNS chunk on its own, with the exception for the first SOA. In that case, the SOA + RRSIG (= 2 RRs) are in the same chunk:

```
darilion@sb34:~$ dig @83.136.34.30 tsigtest.klaus axfr -y "hmac-sha256:..."

; <<>> DiG 9.16.37-Debian <<>> @83.136.34.30 tsigtest.klaus axfr -y hmac-sha256:...
; (1 server found)
;; global options: +cmd
tsigtest.klaus.         30      IN      SOA     sectest1-tst.rcode0.net. rcodezero-soa.ipcom.at. 2023053084 10800 3600 604800 30
tsigtest.klaus.         30      IN      RRSIG   SOA 13 2 30 20230511000000 20230420000000 55768 tsigtest.klaus. eCZK3rD+NtBhHHXT1Obr6akuzoKaFhW/yGJml+2b2gftxZ9wsOxNLsVD fZg7W9L76PuS0S2Lkk+MgLTDyXGNww==
mystigkey.              0       ANY     TSIG    hmac-sha256. 1683112250 300 32 pPX+XyqSwbL2TB9bKrBzTGKTOTcDTABILFlJ+1TS0cs= 47701 NOERROR 0
tsigtest.klaus.         30      IN      NSEC3PARAM 1 0 0 -
mystigkey.              0       ANY     TSIG    hmac-sha256. 1683112250 300 32 cPRixz2WZ3g1kk8SYYm3CVcbf5NpVg5p1AblgDVCQcQ= 47701 NOERROR 0
tsigtest.klaus.         30      IN      RRSIG   NSEC3PARAM 13 2 30 20230511000000 20230420000000 55768 tsigtest.klaus. 9ZknXlnhXJplpNlo8gRyeU0ffqkcUAniahFvuN3mfgjT5QKBWL2nsFU+ stwLhaEhm0SElWwnGm4uIMIyZOOw0A==
mystigkey.              0       ANY     TSIG    hmac-sha256. 1683112250 300 32 dV8XNFT3XYhDK9TCZeU2V8u+QhO3Zq7DNFoYfNTBgLo= 47701 NOERROR 0
tsigtest.klaus.         30      IN      DNSKEY  256 3 13 gD7qcM0fzDnxXhvkYyjU1mh9AvEdfr31GZaHTT5QTBLaZJG67nE7xxsV Lw6acyHd1BEv9BE62KkeS7TaLzaNRQ==
mystigkey.              0       ANY     TSIG    hmac-sha256. 1683112250 300 32 47WoorK2LjRXpJFHg4iLENA9VqrnFhtnplpewOM7YEM= 47701 NOERROR 0
tsigtest.klaus.         30      IN      DNSKEY  257 3 13 A1y2fZ3VUcfSbZy0RvOD3j7Kr72ACLiq/kMX9xYDkN9qh5iAZn2umcbv pkpFzsgtmbP6Rm91nrzmrRTFXGtc1g==
mystigkey.              0       ANY     TSIG    hmac-sha256. 1683112250 300 32 vS7ZmPqc4dqJEF0YQaOPsYSaSb07gEsxqRuPJsz24N0= 47701 NOERROR 0
tsigtest.klaus.         30      IN      RRSIG   DNSKEY 13 2 30 20230511000000 20230420000000 46600 tsigtest.klaus. +9by00vZb0jMhchazrfMYp7gfJduZ79G5b/W/DdaPi+tiAWnjTFP9jdy oiRi4oUkJCFdMwLU34R8WHAjLONmFw==
mystigkey.              0       ANY     TSIG    hmac-sha256. 1683112250 300 32 vuY4NWMQR3kXFOnmdy4FhORcLVYDSipk2U1jLanbWsk= 47701 NOERROR 0
tsigtest.klaus.         30      IN      NS      sectest1-tst.rcode0.net.
mystigkey.              0       ANY     TSIG    hmac-sha256. 1683112250 300 32 rhQ0XDJg49EjKLXrg9yj3v+Fn5KNAnTTJuI4xqbFtN8= 47701 NOERROR 0
tsigtest.klaus.         30      IN      NS      sectest2-tst.rcode0.net.
mystigkey.              0       ANY     TSIG    hmac-sha256. 1683112250 300 32 w4NDcoNxpQgAfw95q0ff0WlMHKLQ+VvGl0avG+eWPMI= 47701 NOERROR 0
tsigtest.klaus.         30      IN      RRSIG   NS 13 2 30 20230511000000 20230420000000 55768 tsigtest.klaus. tSwuWSAyhs/RvSswjCVNJW3JGJsY93pqXNGOLaGtckIKAQ+Je6XJ20iN 2xOZKkzWIp2Ksd8AHvwFFd+TfoLRZw==
mystigkey.              0       ANY     TSIG    hmac-sha256. 1683112250 300 32 hupNlBz6DdKSjMRVpfFjonALQyoH1e9FKUUYJv553xs= 47701 NOERROR 0
i650r0693vi3dmcqguom2elqa4qh42d5.tsigtest.klaus. 30 IN NSEC3 1 0 0 - I650R0693VI3DMCQGUOM2ELQA4QH42D5 NS SOA RRSIG DNSKEY NSEC3PARAM
mystigkey.              0       ANY     TSIG    hmac-sha256. 1683112250 300 32 MWuyeCV46toBBO7YlgQBVBRaWooBLh/okqCjtYwgRQo= 47701 NOERROR 0
i650r0693vi3dmcqguom2elqa4qh42d5.tsigtest.klaus. 30 IN RRSIG NSEC3 13 3 30 20230511000000 20230420000000 55768 tsigtest.klaus. cHTpAeDZPm8VwYMH6tKX91smdfIy0WIke9nVOGXDN47wVhaw8OqQgbRB YGSGUdUqmUxum5ASOqY1grgo/xB+NQ==
mystigkey.              0       ANY     TSIG    hmac-sha256. 1683112250 300 32 TjWvlqqB/jSeziwNs/Nbh90tTqilSHV2L5c7j67ZN3U= 47701 NOERROR 0
tsigtest.klaus.         30      IN      SOA     sectest1-tst.rcode0.net. rcodezero-soa.ipcom.at. 2023053084 10800 3600 604800 30
mystigkey.              0       ANY     TSIG    hmac-sha256. 1683112250 300 32 ltL7AbC5y33Im379LkRSlv4EfU/W6E0cvLL1VZKEIQ8= 47701 NOERROR 0
;; Query time: 91 msec
;; SERVER: 83.136.34.30#53(83.136.34.30)
;; WHEN: Wed May 03 13:10:50 CEST 2023
;; XFR size: 13 records (messages 12, bytes 2576)
```

**Comment by Habbie** [comment]
If you can make this a switch (so the choices are limited to 100 and 1), I'd love to ship this in 4.8.

**Comment by zeha** [comment]
Its a switch now.

File: docs/settings.rst [hunk 1]
```diff
@@ -1982,6 +1982,20 @@ If the webserver should print arguments.
 
 If a PID file should be written.
 
+.. _setting-workaround-11804:
+
+``workaround-11804``
+-------------------
```
Comment (Habbie) [inline docs/settings.rst:1988]: WARNING: Title underline too short.

Comment (Habbie) [inline docs/settings.rst:1988]:
```suggestion
--------------------
```

File: docs/settings.rst [hunk 2]
```diff
@@ -1982,6 +1982,20 @@ If the webserver should print arguments.
...
+
+-  Boolean
+-  Default: no
+
+Workaround for issue https://github.com/PowerDNS/pdns/issues/11804.
```
Comment (Habbie) [inline docs/settings.rst:1993]:
```suggestion
Workaround for `issue #11804 (outgoing AXFR may try to overfill a chunk and fail) <https://github.com/PowerDNS/pdns/issues/11804>`_.
```

**Review by Habbie — CHANGES_REQUESTED** [review]
doc nits, otherwise good!

File: docs/settings.rst (hunk 1 above)
Comment (zeha) [reply docs/settings.rst]: Applied, thanks

File: docs/settings.rst (hunk 2 above)
Comment (zeha) [reply docs/settings.rst]: Applied, thx

**Review by Habbie — APPROVED** [review]

**Pull request merged**